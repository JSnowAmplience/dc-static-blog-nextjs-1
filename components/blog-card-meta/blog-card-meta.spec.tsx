/* eslint-env jest */
import renderer from 'react-test-renderer';
import BlogCardMeta from './blog-card-meta';
import { MediaType } from '../../common/interfaces/media.interface';

describe('BlogCardMeta', () => {
  test('renders full blog card meta', async () => {
    const props = {
      authors: [
        {
          name: 'test-name',
          avatar: {
            image: {
              id: 'image-id',
              name: 'image-name',
              endpoint: 'image-endpoint',
              defaultHost: 'image-default-host',
              mediaType: MediaType.IMAGE
            },
            altText: ''
          }
        }
      ],
      publishedDate: '2019-08-13'
    };
    const component = await renderer.create(<BlogCardMeta {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('renders full blog card meta with multiple authors', async () => {
    const props = {
      authors: [
        {
          name: 'test-name',
          avatar: {
            image: {
              id: 'image-id',
              name: 'image-name',
              endpoint: 'image-endpoint',
              defaultHost: 'image-default-host',
              mediaType: MediaType.IMAGE
            },
            altText: ''
          }
        },
        {
          name: 'test-name-2',
          avatar: {
            image: {
              id: 'image-id-2',
              name: 'image-name-2',
              endpoint: 'image-endpoint-2',
              defaultHost: 'image-default-host-2',
              mediaType: MediaType.IMAGE
            },
            altText: ''
          }
        }
      ],
      publishedDate: '2019-08-13'
    };
    const component = await renderer.create(<BlogCardMeta {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
