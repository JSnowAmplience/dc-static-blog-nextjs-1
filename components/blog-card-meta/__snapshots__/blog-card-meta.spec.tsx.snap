// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCardMeta renders full blog card meta 1`] = `
<div
  className="jsx-2519490847 card-meta"
>
  <span
    className="jsx-2519490847"
  >
    by 
    test-name
  </span>
  <span
    className="jsx-2519490847 publish-date"
  >
    13 August, 2019
  </span>
</div>
`;
